layer.md

```diff
@@ -186,6 +186,38 @@ rootfs-c9d-v1.s1/
         my-app-tools
 ```
 
+### Symbolic Links
+
+When applying a changeset, (assessing the changeset, or creating the original comparison filesystem) symlinks that are either:
+
+- Already existing in a previous layer
+- Added in the current layer
+
+SHOULD NOT be followed. 
+If the path of a changeset contains a symlink component, the symlink component MUST be removed and replaced with a directory as well as any necessary subdirectories to reconstruct the path of the changeset. 
+If there are changeset entries for any of these directories, the corresponding changeset MUST still be applied on those directories so that they have the correct metadata. If there is not a changeset entry for a directory created in this manner, the metadata of the created directory is implementation-defined but implementations SHOULD strive to make the results consistent. 
+Implementations SHOULD NOT produce changesets which have paths that contain symlink components.
+
+However, an implementation MAY choose to provide a means to allow the user to explicitly ask to follow symlink path components.
+As an example, umoci provides a flag `--keep-dirlinks`. 
+Given such a directive to follow symlinks, such links SHOULD only be followed within the scope of the root of the comparison filesystem.
+
+Implementations SHOULD NOT restrict the target of a symlink, because symlinks are not resolved at linking time and thus adding restrictions on the target of a symlink is not necessary -- the container process will be restricted in its ability to resolve paths outside of the container root filesystem.
+
+Link replacement in a later layer SHOULD be replaced, following the same convention as a changeset over existing files.
```

Comment:
this sentence is not clear to me

Reply:
I think it's saying if we find a link replacement in some layer after one we've parsed, we should honor it. How should we say that so it's clear?

Reply:
It might just be implied this happens and this is too much detail, so I can also take it out. cc @cyphar and @reidpr

Reply:
It's a little fuzzy for me too but symlinks are so tricky I'm in favor of being explicit, even if in some kind of informal note.

Reply:
okay I'm not sure what it's supposed to say - maybe @cyphar can add insight.